stacker/plan.py

```diff
@@ -182,14 +191,30 @@ def _single_run(self):
                 )
                 continue
 
+            if not step.done and self._watch_func and step_name not in self._watchers:
```

Comment:
This seems like it could use some tests.  A little nervous about introducing multiprocessing without any.